cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1257,6 +1262,66 @@ def make_resource(self, data, **kwargs):
         return HeadNode(**data)
 
 
+class LoginNodeImageSchema(BaseSchema):
+    custom_ami = fields.Str(validate=validate.Regexp(r"^ami-[0-9a-z]{8}$|^ami-[0-9a-z]{17}$"))
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodeImage(**data)
+
+
+class LoginNodeSshSchema(BaseSchema):
+    key_name = fields.Str(required=True)
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodeSsh(**data)
+
+
+class LoginNodeNetworkingSchema(BaseNetworkingSchema):
+    subnet_id = fields.Str(required=True)
+    elastic_ip = fields.Str()
+    proxy = fields.Str()
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return LoginNodeNetworking(**data)
+
+
+class LoginNodePoolSchema(BaseSchema):
+    name = fields.Str(required=True, validate=[validate.Length(min=1), validate.Regexp(r'^[a-z0-9]+$')])
```

Comment:
The Length=1 validation should probably be in the `LoginNodesSchema`.
Also, let's remove this regexp for now (to be consistent with other naming checks).

Reply:
I added Length(min=1) validation because I thought the length of the 'name' string should be longer than 0. But I have removed it. And also removed the regexp.